src/Process/PointCloudProcessing.js

```diff
@@ -293,22 +228,52 @@ export default {
 
         layer.displayedCount = 0;
         for (const pts of layer.group.children) {
-            layer.displayedCount += pts.geometry.drawRange.count;
+            if (pts.material.visible) {
+                const count = pts.geometry.attributes.position.count;
+                pts.geometry.setDrawRange(0, count);
+                layer.displayedCount += count;
+            }
         }
 
         if (layer.displayedCount > layer.pointBudget) {
-            const reduction = layer.pointBudget / layer.displayedCount;
-            for (const pts of layer.group.children) {
-                if (pts.material.visible) {
-                    const count = Math.max(1.0, Math.floor(pts.geometry.drawRange.count * reduction));
-                    pts.geometry.setDrawRange(0, count);
+            // 2 different point count limit implementation, depending on the pointcloud source
+            if (layer.metadata.customBinFormat) {
```

Comment:
I would rather have the processing be agnostic of formats, nut depend rather on format properties or strategies. Could not you have reordered bin files as in cin files, or the other way around?

Reply:
What about renaming customBinFormat to something like prefixSubsampling ?

Reply:
I added a `layer.supportsProgressiveDisplay` property. (Note that the rest of the file relies on the potreeconverter format, there's more work to do to be able to use this processing code on any pointcloud source)